
 backport of upstream patch:

	Commit: e143d1f00497f0178a1febec8fb4aa7c842fa35a
	From: Karel Zak <kzak@redhat.com>
	Date: Mon, 12 Jul 2010 11:57:55 +0200
	Subject: [PATCH] aggety: don't wipe CLOCAL flag

	gettey should not clear the flag. It seems that the flag is
	automatically enabled for serial consoles tht don't use CD signal.

	Addresses: http://bugzilla.redhat.com/show_bug.cgi?id=598631
	Signed-off-by: Karel Zak <kzak@redhat.com>

diff -up util-linux-ng-2.18/login-utils/agetty.c.kzak util-linux-ng-2.18/login-utils/agetty.c
--- util-linux-ng-2.18/login-utils/agetty.c.kzak	2010-10-22 21:51:31.000000000 +0200
+++ util-linux-ng-2.18/login-utils/agetty.c	2010-10-22 21:51:31.000000000 +0200
@@ -715,7 +715,7 @@ termio_init(tp, speed, op)
 
     tp->c_iflag = tp->c_lflag = tp->c_oflag = 0;
 
-    tp->c_cflag = CS8 | HUPCL | CREAD;
+    tp->c_cflag = CS8 | HUPCL | CREAD | (tp->c_cflag & CLOCAL);
 
     /* Note that the speed is stored in the c_cflag termios field, so we have
      * set the speed always when the cflag se reseted.
